mitmproxy/proxy/context.py

```diff
@@ -118,6 +122,15 @@ def __init__(self, peername, sockname, timestamp_start):
         self.sockname = sockname
         self.timestamp_start = timestamp_start
 
+    def __str__(self):
+        if self.alpn:
+            tls_state = f", alpn={self.alpn.decode(errors='replace')}"
```

Comment:
pedantic nitpick from previous discussion, again we probably will never encounter this:
instead of replacing or hiding characters, it would make more sense to either display ASCII (if everything is ASCII), or the hex bytes.

Reply:
This function is only every visible when debugging, so we should be good here.